design/auth.md

```diff
@@ -0,0 +1,270 @@
+
+### Search
+
+Pipeline List API will _only_ return "publicly accessible" pipelines.  That means "private" repositories will not show up in search.  Additionally, if Screwdriver is in "Required Login" mode, then that API will still return "publicly accessible" repositories, but require a valid user Bearer token.
+
+### Resources
+
+The resources (or scope) that we care about are `pipelines`, their `jobs`, and their `builds`.  There are two types of permission, `read` and `write`.
+
+| **type** | **scope syntax** | **read** | **write** |
+| -------- | ---------------- | -------- | --------- |
+| `pipeline` | `pipeline:(?<pipeline_id>\d+)` | can view details about the pipeline | can create, edit, and delete the pipeline |
+| `job` | `job:(?<job_id>\d+)` | can view details about the job, its builds, and artifacts produced from the build | can start, stop, enable, and disable the job |
+| `build` | `build:(?<build_id>\d+)` | can view details about the build and artifacts produced from the build | can update status, log lines, and publish artifacts about a build |
+
+#### Inheritance
+
+- Read permissions inherit bi-directionally (and do not need to be included).
+- Write permissions automatically include read permission.
+
+This inheritance is designed to reduce the amount of content that needs to be stored in the JWT.
+
+#### Examples
+
+##### 1 - Owner
+
+If `jane` requests access to a pipeline `20` and `jane` is an owner of the repository:
+
+```json
+{
+  ....
+  "scope": [
+    "project:20": "write",
+    "job:100": "write",
+    "job:101": "write",
+    "job:102": "write",
+    "job:103": "write"
+  ]
+}
+```
+
+_Note, in this example `builds` are not listed because it's assumed they have read permission based on the `jobs` or `projects` listed._
+
+##### 2 - Collaborator
+
+If `bob` requests access to the `20` pipeline and `bob` is just a collaborator of the repository:
+
+```json
+{
+  ....
+  "scope": [
+    "project:20": "read",
+    "job:100": "write",
+    "job:101": "write",
+    "job:102": "write",
+    "job:103": "write"
+  ]
+}
+```
+
+##### 3 - Read-Only
+
+If `mal` requests access to the `20` pipeline and `mal` just has read access to the repository:
+
+```json
+{
+  ....
+  "scope": [
+    "project:20": "read"
+  ]
+}
+```
+
+##### 4 - Pull-Request Submitter
+
+If `pat` requests access to the `20` pipeline, the `20` project is public, `pat` is not a collaborator, and `pat` opened `PR-15` (ID `103`):
+
+```json
+{
+  ....
+  "scope": [
+    "project:20": "read",
+    "job:103": "write"
+  ]
+}
+```
+
+##### 5 - No Access
+
+If `sue` requests access to the `21` pipeline, the `21` project is private, `sue` is not a collaborator:
+
+```json
+404 Resource Not Found
+```
+
+_Note, the reason we return 404 is to mask private repositories._
```

Comment:
Should add in a note that this is how Github (and maybe others) operate? Might be an unnecessary detail, but I thought it could be good to show that this is standard industry practice